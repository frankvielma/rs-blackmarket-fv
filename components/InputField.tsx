export default function InputField({
  type = '',
  id = '',
  label = '',
  placeholder = '',
}) {
  return (
    <div className="my-4 flex flex-col">
      <label htmlFor={type} className="text-sm md:text-base">
        {label}
      </label>
      <input
        id={id}
        placeholder={placeholder}
        className="custom-focus h-[34px] w-auto rounded-md border border-dark-violet ps-3 placeholder:text-base placeholder:text-dark-gray hover:border-hover active:outline-none active:ring-2 active:ring-offset-2 md:h-[44px]"
        type={type}
      />
    </div>
  )
}
