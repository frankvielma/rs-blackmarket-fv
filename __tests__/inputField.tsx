import { render, fireEvent, screen } from '@testing-library/react'
import InputField from '@/components/InputField'

describe('inputField', () => {
  it('renders inputField unchanged', () => {
    const { container } = render(<InputField />)
    expect(container).toMatchSnapshot()
  }),
    it('calls the onChange function', () => {
      render(<InputField label={'Email'} type={'email'} id={'email'} />)
      const inputElement = screen.getByLabelText('Email')
      fireEvent.change(inputElement, { target: { value: 'test value' } })
      expect((inputElement as HTMLInputElement).value).toBe('test value')
    }),
    it('calls the handleBlur function', () => {
      render(<InputField label={'Email'} type={'email'} id={'email'} />)
      const inputElement = screen.getByLabelText('Email')
      fireEvent.blur(inputElement)
      expect(inputElement).toHaveClass(
        'border border-red-600 custom-focus h-[34px] w-auto rounded-md border border-dark-violet ps-3 placeholder:text-base placeholder:text-dark-gray hover:border-hover active:outline-none active:ring-2 active:ring-offset-2 md:h-[44px]'
      )
    })
})
