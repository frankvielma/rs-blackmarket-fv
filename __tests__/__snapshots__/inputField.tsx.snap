// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputField renders inputField unchanged 1`] = `
<div>
  <div
    class="my-4 flex flex-col"
  >
    <label
      class="text-sm md:text-base"
      for=""
    />
    <input
      class="border custom-focus h-[34px] w-auto rounded-md border border-dark-violet ps-3 placeholder:text-base placeholder:text-dark-gray hover:border-hover active:outline-none active:ring-2 active:ring-offset-2 md:h-[44px]"
      id=""
      placeholder=""
      required=""
      type=""
    />
    <span />
  </div>
</div>
`;
